viewing_party/party.py

```diff
@@ -0,0 +1,204 @@
+import operator
+
+# Wave 1 
+def create_movie(title, genre, rating):
+    if title and genre and rating:
+        movie_to_watch = {
+                "title" : title,
+                "genre" : genre,
+                "rating" : rating 
+                }    
+        return movie_to_watch      
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    list_of_watchlist = user_data["watchlist"]
+    for movie_to_watch in list_of_watchlist:
+        if movie_to_watch["title"] == title:
+            add_to_watched(user_data, movie_to_watch)
+            list_of_watchlist.remove(movie_to_watch)
```

Comment:
even though remove works great here, be careful using it in a for loop. It can cause side effects. Here is a link that explains it more with examples https://thispointer.com/python-remove-elements-from-a-list-while-iterating/